lib/tasks/errors.rake

```diff
@@ -15,32 +15,43 @@ namespace :errors do
 
   task :fetch_log => :environment do
     request_id = ENV['request_id']
-    logfile = ENV['log_file'] || 'production.log' # default file where we want to find is production.log
     unless request_id
       puts "Can't find log without request_id"
       exit(1)
     end
+
     type = Foreman::Logging.config[:type]
-    unless type == 'file'
-      puts "This rake task only support log file, you're using #{type}"
-      puts "You can search the logs in #{type} for request_id #{request_id}"
-      exit(1)
-    end
-    if Foreman::Logging.config[:layout] != 'multiline_request_pattern'
-      puts "Warning: Logging layout is not multiline_request_pattern."
-      puts "This output of this command can be incomplete."
-    end
-    file_path = File.join(Foreman::Logging.log_directory, logfile)
-    unless File.exist?(file_path)
-      puts "Can't find log file #{file_path}"
-      exit(1)
-    end
-    result = `grep "#{request_id}" "#{file_path}"`
-    if result.empty?
-      puts "Can't find log for request_id"
-      exit(1)
-    end
-    puts result
+    lines = case type
+            when 'file'
+              logfile = ENV['log_file'] || 'production.log' # default file where we want to find is production.log
+              if Foreman::Logging.config[:layout] != 'multiline_request_pattern'
+                puts "Warning: Logging layout is not multiline_request_pattern."
+                puts "This output of this command can be incomplete."
+              end
+              file_path = File.join(Foreman::Logging.log_directory, logfile)
+              unless File.exist?(file_path)
+                puts "Can't find log file #{file_path}"
+                exit(1)
+              end
+              result = `grep "#{request_id}" "#{file_path}"`
+              if result.empty?
+                puts "Can't find log for request_id"
+                exit(1)
+              end
+              puts result
+            when 'journald'
+              lines = `journalctl "REQUEST=#{request_id}"`
```

Comment:
Just a thought: perhaps we can catch the permission error and tell users to run the command manually?